spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -201,7 +197,19 @@ def env(self):
             env_vars['SPY_SYMPY_O'] = False
             env_vars['SPY_RUN_CYTHON'] = True
 
+        # internal interpreter and external interpreter have mutually exclusive
+        # settings
+        if default_interpreter and os.environ.get('PYTHONHOME'):
+            # default interpreter should keep PYTHONHOME
+            env_vars.update({'PYTHONHOME': os.environ.get('PYTHONHOME')})
+        elif not default_interpreter and os.environ.get('HOME'):
+            # Add HOME to env_vars
+            env_vars.update({'HOME': os.environ.get('HOME')})
+
         # Add our PYTHONPATH to env_vars
+        # spyder env python should be omitted in kernels
```

Comment:
why is this the case?

Reply:
This refers to the PYTHONPATH variable in Spyder's launch environment. When starting a new iPython kernel, we don't want to inherit the PYTHONPATH from Spyder's launch environment, which could give access to python packages the user does not expect and pollutes the PYTHONPATH. Or worse: accessing the _same_ packages of a different version unwittingly. I think the PYTHONPATH should be comprised _only_ of those explicitly provided by the user in the Python Path Manager and possibly a user's login shell (~/.bash_profile, etc.).

Reply:
So in this case it sounds like this is a bug in spyder and not a MacApp thing...

Reply:
In this specific case, I believe so. I think the Mac app effort has just revealed few "cross-environment" issues.

Reply:
in that case this needs a test ;-)

Reply:
Okay, I can work on a test...

Reply:
Oh, I forgot that I had already modified `test_preserve_pypath` in this PR. Is that what you're looking for?